### ISSUE TITLE
iPad air silent fail, works on android, iOS iPhone SE (2 gen)

### ISSUE THREAD

**Issue by skionthursday** [Baseflow/flutter-geolocator#480]
## 🐛 Bug Report

final Position _position = await Geolocator()
        .getCurrentPosition(desiredAccuracy: LocationAccuracy.high);
    _latitudeController.text = _position.latitude.toString();
    _longitudeController.text = _position.longitude.toString();
    _altitudeController.text = _position.altitude.toString();

In a flutter solution, I run the above code to get values to populate a screen. Above fails with no message or error on a iPad air. Remains steps after the above are not executed causing the screen functions to fail.

### Expected behavior

Should work as expected as on other platforms.  Currently working on Android solutions and iOS simulators

### Reproduction steps

Using Flutter and Visual Studio code, deploy to iOS simulator and it works.  Plug in actual iPad Air and run with debugging and you should see the function exit with no message after the first line above attempts to execute.

### Configuration

Running:

Flutter 1.17.5
VS Code 1.47.3 (June 2020)
GeoLocation 5.3.2+2
iPad Air model 1474 / MD785LL/B  -- running iOS 12.4.8
iOS Simulator iPhone SE (2nd gen)

Running Flutter, VS Code, etc. on MacBook running MacOS Catalina 10.15.6

**Version:** 1.x

**Platform:** 
- [x] :iphone: iOS
- [ ] :robot: Android

**Comment by skionthursday** [comment]
Additional diagnostics information:  When deploying to an iPad Air Gen 3 simulator, no problem as geolocation works.

Debugging the code, I preface the problem with PermissionHandler().checkPermissionStatus(PermissionGroup.locationWhenInUse) --> returns PermissionStatus.granted

Starting from scratch, I start my app, request and grant permission. Then the app starts and I do not call geolocation.  I reboot the iPad and check privacy and I do see allow location access set to be While Using the App is checked. I restart the app and PermissionStatus.granted continues, but fails as described above when executing await Geolocator().getCurrentPosition(desiredAccuracy: LocationAccuracy.high);

regards, Bob

**Comment by skionthursday** [comment]
<img width="767" alt="Screen Shot 2020-08-02 at 4 00 18 PM" src="https://user-images.githubusercontent.com/66576919/89140172-1659ff80-d4fe-11ea-8775-8086256d9e5f.png">

Some additional diagnostics: Reinstalling, and responding to the location permission request -- when responding "Always" the failure is silent.  When responding allow to "Use location when using app", I have been able to try/catch an error message:  PlatformExeception(PERMISSION_DENIED, Access to location data denied, null) which is interesting a did agree/allowed permission.

**Comment by skionthursday** [comment]
Greeting from Colorado, USA!

I am sorry to report that 6.0.0-rc.1 did not fix the problem and that the failing behavior is the same. Works on iPhone Se simulator, silent fail on the physical iPad.  Let me know if there is anything else I can do to help.

Regards,

Bob

**Comment by mvanbeusekom** [comment]
Hello @skionthursday,

What type of iPad Air are you testing on? The important part is if it is the WiFi only version or if it also has a sim-card slot? The WiFi only version doesn't have a GPS chip onboard and would make sense that it won't return any location information. If this is the case I can have a look if Apple provides an API to detect this and have the geolocator return and exception in this case.

Regards from the Netherlands.

**Comment by skionthursday** [comment]
Hello Maurits,

Yes this is an iPad Air, no sim-card WIFI only.  Model MD785LL/B, SN# DMTPCHVEFK10, Software version 12.4.8.  I downloaded the app from the app store: GPS Coordinates v 3.4.1 and captured the attached screen shot showing.  The lack of accuracy has me thinking that is location was derived from WIFI location.  I turned off the WIFI and rebooted the iPad and no location was found. I turned on the WIFI and still no location was found and never returns. Interesting as I’ve seen this one-time reporting before on this device and could not explain it. So I think your thoughts on what is happening are correct.  

With the silent fail and dumping out of the routine – I have this work around.  I have two routines. The first sets a database file field to be false. Then I use Geolocation to get latitude and if successful I set the database field to true.  If it fails to get the latitude, the routine fails, exits and the database field does not get updated and remains false.  

In my second routine which is in my main app flow, I test the database field and it tells me Geolocation status. If the database field is false, I don’t try to use Geolocation and display on my app that “GPS locations is not available on this device” and avoid the routine failing and prematurely exiting.  

I hope this helps.

Regards, Bob
